tests/blockchain/eth/contracts/main/miners_escrow/test_mining.py

```diff
@@ -297,3 +297,181 @@ def test_mining(testerchain, token, escrow_contract):
+    testerchain.wait_for_receipt(tx)
+    assert 280 == escrow.functions.getLockedTokens(ursula).call()
+    assert 370 == escrow.functions.getLockedTokens(ursula, 1).call()
+    assert 370 == escrow.functions.minerInfo(ursula).call()[VALUE_FIELD]
+    assert 370 == escrow.functions.lockedPerPeriod(period + 1).call()
+
+    # After two periods shortest sub stake will be unlocked, lock again and slash after this
+    testerchain.time_travel(hours=1)
+    period += 1
+    assert 370 == escrow.functions.getLockedTokens(ursula).call()
+    assert 270 == escrow.functions.getLockedTokens(ursula, 1).call()
+    assert 100 == escrow.functions.getLockedTokens(ursula, 3).call()
+    assert 0 == escrow.functions.lockedPerPeriod(period - 1).call()
+    assert 370 == escrow.functions.lockedPerPeriod(period).call()
+    assert 0 == escrow.functions.lockedPerPeriod(period + 1).call()
+    tx = escrow.functions.lock(100, 2).transact({'from': ursula})
+    testerchain.wait_for_receipt(tx)
+    assert 370 == escrow.functions.getLockedTokens(ursula).call()
+    assert 370 == escrow.functions.getLockedTokens(ursula, 1).call()
+    assert 100 == escrow.functions.getLockedTokens(ursula, 3).call()
+    assert 0 == escrow.functions.lockedPerPeriod(period - 1).call()
+    assert 370 == escrow.functions.lockedPerPeriod(period).call()
+    assert 370 == escrow.functions.lockedPerPeriod(period + 1).call()
+    deposit = escrow.functions.minerInfo(ursula).call()[VALUE_FIELD]  # Some reward is already mined
+
+    # Slash two sub stakes:
+    # one which will be unlocked after current period and new sub stake
+    tx = overseer.functions.slashMiner(ursula, 10, investigator, 0).transact()
+    testerchain.wait_for_receipt(tx)
+    assert 360 == escrow.functions.getLockedTokens(ursula).call()
+    assert 360 == escrow.functions.getLockedTokens(ursula, 1).call()
+    assert 100 == escrow.functions.getLockedTokens(ursula, 3).call()
+    assert deposit - 10 == escrow.functions.minerInfo(ursula).call()[VALUE_FIELD]
+    assert 0 == escrow.functions.lockedPerPeriod(period - 1).call()
+    assert 360 == escrow.functions.lockedPerPeriod(period).call()
+    assert 360 == escrow.functions.lockedPerPeriod(period + 1).call()
+
+    # Slash three sub stakes:
+    # one which will be unlocked after current period, new sub stake and another short sub stake
+    tx = overseer.functions.slashMiner(ursula, 90, investigator, 0).transact()
+    testerchain.wait_for_receipt(tx)
+    assert 270 == escrow.functions.getLockedTokens(ursula).call()
+    assert 270 == escrow.functions.getLockedTokens(ursula, 1).call()
+    assert 100 == escrow.functions.getLockedTokens(ursula, 3).call()
+    assert deposit - 100 == escrow.functions.minerInfo(ursula).call()[VALUE_FIELD]
+    assert 0 == escrow.functions.lockedPerPeriod(period - 1).call()
+    assert 270 == escrow.functions.lockedPerPeriod(period).call()
+    assert 270 == escrow.functions.lockedPerPeriod(period + 1).call()
```

Comment:
I think these tests will need, at some moment, some touch ups. I don't find immediate what do the amounts in the assertions mean. Perhaps this can be described better, although I'm afraid I don't have any suggestions right now.